sklearn/metrics/cluster/supervised.py

```diff
@@ -716,9 +719,8 @@ def adjusted_mutual_info_score(labels_true, labels_pred, max_n_classes=5000):
     if (classes.shape[0] == clusters.shape[0] == 1 or
             classes.shape[0] == clusters.shape[0] == 0):
         return 1.0
-    contingency = contingency_matrix(labels_true, labels_pred,
-                                     max_n_classes=max_n_classes)
-    contingency = np.array(contingency, dtype='float')
+    contingency = contingency_matrix(labels_true, labels_pred, sparse=True)
+    contingency = contingency.astype(float)
```

Comment:
`.astype(float)` is not guaranteed to use the same level of precision on all platforms. To get platform agnostic results it's better to be specific:

```python
contingency = contingency.astype(np.float64)
```